tensorlayer/cli/train.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+# example usage:
+#   ./tensorlayer/cli/train.py ./example/tutorial_mnist_distributed.py
+#   ./tensorlayer/cli/train.py ./example/tutorial_imagenet_inceptionV3_distributed.py -- --batch_size 16
+
+import argparse
+import json
+import multiprocessing
+import os
+import re
+import subprocess
+import sys
+
+PORT_BASE = 10000
+
+
+def _get_gpu_ids():
```

Comment:
I this code is much better and also takes into account CUDA_VISIBLE_DEVICES:
```
from tensorflow.python.client import device_lib

def get_available_gpus():
    local_device_protos = device_lib.list_local_devices()
    return [x.name for x in local_device_protos if x.device_type == 'GPU']
```

Reply:
Hi Jorge, we were using this piece of the code to get GPUs before but it actually causes problems in tensorflow initialization so we switch to this approach temporarily.